numpy/lib/npyio.py

```diff
@@ -599,13 +611,41 @@ def savez_compressed(file, *args, **kwds):
     """
     _savez(file, args, kwds, True)
 
+@contextlib.contextmanager
```

Comment:
Nb that that's not the right way to write a context manager -- if the with
block raises an exception then the yield will raise that exception and
os.remove won't be called, you still need a try/finally.
On Nov 5, 2015 2:03 PM, "Charles Harris" <notifications@github.com> wrote:

> In numpy/lib/npyio.py
> <https://github.com/numpy/numpy/pull/6545#discussion_r44076176>:
>
> > @@ -599,13 +611,41 @@ def savez_compressed(file, *args, **kwds):
> >      """
> >      _savez(file, args, kwds, True)
> >
> > +@contextlib.contextmanager
>
> Maybe something simpler?
>
> @contextmanager
> def TempFileName(suffix='', prefix='tmp', dir=None):
>     fid, name = mkstemp(suffix, prefix, dir)
>     os.close(fid)
>     yield name
>     os.remove(name)
>
> This is a common pattern in a lot of tests, e.g, in test_io.py
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/6545/files#r44076176>.
>